### DISCUSSION TITLE
Buy / Economy events

### DISCUSSION THREAD

**Discussion by n1c** [saul/demofile#349] [Questions] [answered]
Hello!

From what I can see there isn't an "easy" way of getting player buy events is this correct?

I'm kind of assuming I'll have to do something weird like watch for differences in player `cashSpendThisRound` and `weapons` to try piece together things they've bought?

Thanks in advance!

**Comment by saul** [comment, marked as answer]
Take a look at the [purchases](https://github.com/saul/demofile/blob/master/examples/purchases.ts) example. Specifically the bit you need is:

```ts
demoFile.entities.on("create", e => {
  const weapon = e.entity;
  if (!(weapon instanceof Weapon)) return;

  const owner = weapon.owner;
  if (!owner) return;

  // Skip over weapons that are given and not bought
  if (
    weapon.serverClass.name === "CKnife" ||
    weapon.serverClass.name === "CC4" ||
    (owner.teamNumber === 2 && weapon.serverClass.name === "CWeaponGlock") ||
    (owner.teamNumber === 3 && weapon.serverClass.name === "CWeaponHKP2000")
  )
    return;

  console.log(
    `'${owner.name}' just bought a '${weapon.itemName}' (${weapon.quality})`
  );
});
```

**Reply by n1c** [reply]
Ah gosh, it's been so long since I used this library I should really have started at the `examples` folder!

Thanks so much Saul.
